hpx/runtime/parcelset/parcelport_impl.hpp

```diff
@@ -913,6 +913,7 @@ namespace hpx { namespace parcelset
             else
             {
                 ++operations_in_flight_;
+                // NOLINTNEXTLINE(bugprone-use-after-move)
                 HPX_ASSERT(num_parcels < parcels.size());
 
```

Comment:
This is a false positive, but I'm wondering if we couldn't avoid it. @sithhell it seems like clang-tidy is complaining because it can't see that the lambda which evaluates the expression is evaluated right away. I'm wondering what the lambda gives us? I don't see immediately why we couldn't just evaluate the expression right away. (I'm talking about this: https://github.com/STEllAR-GROUP/hpx/blob/2d1581bc6a01c47778c4380193b6b6b613d78fb0/libs/assertion/include/hpx/assertion.hpp#L54-L59.)

Reply:
The lambda eliminates unused variable warnings, iirc. I'm a little clueless why this warning appears in this case though...

Reply:
Maybe it did in some earlier version, but in release mode it definitely won't stop unused variable warnings. I think it's a fair, but somewhat paranoid warning. I'll keep it as it is for now.